hub/core/index/index.py

```diff
@@ -274,9 +317,14 @@ def apply(self, array: np.ndarray):
         return array[index_values]
 
     def is_trivial(self):
-        """Checks if an index is equivalent to the trivial slice `[:]`, aka slice(None)."""
+        """Checks if an Index is equivalent to the trivial slice `[:]`, aka slice(None)."""
         return (len(self.values) == 1) and self.values[0].is_trivial()
 
+    def length(self, parent_length: int):
+        """Returns the primary length of an Index given the length of the parent it is indexing.
+        See: IndexEntry.length"""
+        return self.values[0].length(parent_length)
```

Comment:
Can self.values ever be empty and throw an error here?

Reply:
@AbhinavTuli nope. The default empty Index has first value `slice(None)`. It's designed to never be empty so that we can do checks just like this